merkle-tree/concurrent/src/lib.rs

```diff
@@ -155,35 +447,38 @@ where
     /// Increments the changelog counter. If it reaches the limit, it starts
     /// from the beginning.
     fn inc_current_changelog_index(&mut self) {
-        // NOTE(vadorovsky): Apparenty, Rust doesn't have `checked_remainder`
-        // or anything like that.
-        self.current_changelog_index = if MAX_CHANGELOG > 0 {
-            (self.current_changelog_index + 1) % MAX_CHANGELOG as u64
-        } else {
-            0
-        };
+        if self.changelog_capacity > 0 {
+            if self.changelog_length < self.changelog_capacity {
+                self.changelog_length = self.changelog_length.saturating_add(1);
```

Comment:
is saturating really the behavior we want here since it just silently stops incrementing?
([see saturating add](https://doc.rust-lang.org/std/primitive.i8.html#method.saturating_add))
[Checked add](https://docs.rs/num/latest/num/trait.CheckedAdd.html) would be the alternative, it returns None when overflowing.
For example when modifying variables like lamports we should definitely use checked add and fail in case of an overflow.

Reply:
In this case I don't think there is the risk of an overflow since changelog length is a low number. Thus we could also just use unchecked arithmetics.

Reply:
Fair point. I will switch to checked add.